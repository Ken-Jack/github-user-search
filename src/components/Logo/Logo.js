// @flow

import React from 'react';
import {Link} from 'react-router-dom';
import {connect} from 'react-fela';
import GitHubIcon from './mark-github.svg';

type Props = {
  styles: Object,
};

function Logo({styles}: Props) {
  return (
    <div>
      <h1 className={styles.Logo_title}>
        <Link
          to="/"
          className={styles.Logo_link}
        >
          <GitHubIcon
            className={styles.Logo_icon}
            width={24}
            height={24}
          />
          GitHub User Search
        </Link>
      </h1>
      <p className={styles.Logo_subTitle}>
        Browse users and their profiles via <a href="https://developer.github.com/v3/">the GitHub API</a>
      </p>
    </div>
  );
}

const styles = {
  Logo_icon: () => ({
    marginRight: '8px',
  }),

  Logo_title: () => ({
    fontSize: '25px',
  }),

  Logo_link: () => ({
    display: 'flex',
    textDecoration: 'none',
    color: 'inherit',
  }),

  Logo_subTitle: () => ({
    marginTop: '5px',
    fontSize: '14px',
  }),
};

export default connect(styles)(Logo);
