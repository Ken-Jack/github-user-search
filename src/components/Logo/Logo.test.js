import React from 'react';
import felaSnapshot from 'test-util/fela-snapshot';
import {
  StaticRouter as Router,
} from 'react-router-dom';

import Logo from './Logo';

describe('Component: Logo', () => {

  it('should render', () => {
    const component = felaSnapshot(
      <Router context={{}}>
        <Logo />
      </Router>
    );
    expect(component).toMatchSnapshot();
  });

});
