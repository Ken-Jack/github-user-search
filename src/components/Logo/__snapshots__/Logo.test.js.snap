// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Logo should render 1`] = `
Object {
  "component": <div>
    <h1
      className="b"
    >
      <a
        className="c d e"
        href="/"
        onClick={[Function]}
      >
        <svg />
        GitHub User Search
      </a>
    </h1>
    <p
      className="f g"
    >
      Browse users and their profiles via 
      <a
        href="https://developer.github.com/v3/"
      >
        the GitHub API
      </a>
    </p>
  </div>,
  "styles": "
    .a{margin-right:8px}
    .b{font-size:25px}
    .c{display:flex}
    .d{text-decoration:none}
    .e{color:inherit}
    .f{margin-top:5px}
    .g{font-size:14px}",
}
`;
